### PR TITLE
Finish writing the README

### PR DESCRIPTION
**Pull request by Geod24**

### CODE REVIEW

**Comment by Geod24** [comment]
Ping

File: README.md [hunk 1]
```diff
@@ -19,5 +19,20 @@ allowing to use more convenient syntax (e.g. operator overload) when manipulatin
...
-This module glues together `ECC` and `Hash` to generate `Signature`.
-TODO MORE.
+This module glues together `ECC` and `Hash` to generate signatures.
+In order to generate a signature, both the `Scalar` and the `Point` are used,
+so in order to avoid re-generating the `Point` every time, a convenience `Pair` struct is exposed.
+
+The signature generated is stored in a `Sig` structure, however, to separate the chain's data structures
+from the crypto code, `Sig` can convert to a `Signature` (see `agora.crypto.Types`) via the `toBlob` method.
+
+The main interface for this modules are the various `sign` function, which can be used in
```
Comment (hewison-chris) [inline README.md]:
```suggestion
The main interface for this modules are the various `sign` functions, which can be used in
```

File: README.md [hunk 2]
```diff
@@ -19,5 +19,20 @@ allowing to use more convenient syntax (e.g. operator overload) when manipulatin
...
+a pedestrian way (e.g. `Pair kp = Pair.random(); sign(kp, "Message");`), or in a more complex fashion
+(using a specific `r`, or even different scalar and points altogethers, to allow for multisig schemes).
```
Comment (hewison-chris) [inline README.md]:
```suggestion
(using a specific `r`, or even different scalar and points altogether, to allow for multisig schemes).
```

File: README.md (hunk 1 above)
Comment (kchulj) [inline README.md]:
```suggestion
The main interface for this module are the various `sign` functions, which can be used in
```

**Review by hewison-chris — CHANGES_REQUESTED** [review]
LGTM. Just a misplaced s.

**Review by kchulj — APPROVED** [review]
Nitpick.

**Review by hewison-chris — COMMENTED** [review]
LGTM

**Pull request merged**